documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1024,4 +1029,180 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2005" />R2005 LongRunningResponseStatusCode
+**Output Message**: A '{0}' operation '{1}' with x-ms-long-running-operation extension must have a valid terminal success status code {2}.
+
+**Description**: The allowed response status codes for a long 'delete' operation are "200", "204". The allowed response status codes for a 'post' operation are "200", "201" & "204". The allowed response status codes for a 'put' operation are  "200" & "201".
+
+**Why the rule is important**: This will ensure that the 'delete'/'post'/'put' operations are designed correctly.
+
+**How to fix the violation**: Ensure that the 'delete'/'post'/'put' operations have the allowed response codes.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2019" />R2019 ResourceHasXMsResourceEnabled
+**Output Message**: A 'Resource' definition must have x-ms-azure-resource extension enabled and set to true.
+
+**Description**: A 'Resource' definition must have x-ms-azure-resource extension enabled and set to true.
+
+**Why the rule is important**: This will ensure that the 'Resource' definition is designed correctly.
+
+**How to fix the violation**: Ensure that the 'Resource' definition has x-ms-azure-resource extension enabled and set to true.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2064" />R2064 LROStatusCodesReturnTypeSchema
+**Output Message**: 200/201 Responses of long running operations must have a schema definition for return type. OperationId: '{0}', Response code: '{1}'
+
+**Description**: The '200'/'201' responses of the long running operation must have a schema definition.
+
+**Why the rule is important**: This will ensure that the generated SDK understands the operation and the user could get the correct response of the long running operation.
+
+**How to fix the violation**: Ensure that the '200'/'201' responses of the long running operation has a schema definition.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3007" />R3007 PutGetPatchResponseSchema
```

Comment:
isn't this an RPC violation? If so, it should be in its corresponding section. Please review the rest of the ones you included to make sure they are in the correct section.